### PR TITLE
Writing Mock Tests for PaymentService

### PR DESCRIPTION
**Pull request by Valiusikk**

### CODE REVIEW

File: src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (Evgenikaall) [inline src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java]: wildcard

Comment (Valiusikk) [reply src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java]: Fixed!

Comment (Evgenikaall) [reply src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java]: I can't see this

File: src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    @BeforeEach
+    void setUp() {
+        paymentRepository = new InMemPaymentRepository();
+    }
+
+    @Test
+    void findByIdOfSucceds() {
+        //when
+        paymentRepository.save(TEST_PAYMENT);
```
Comment (Evgenikaall) [inline src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java:31]: it must be under // given

File: src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                () -> paymentRepository.editMessage(test_uuid, message));
+
+        //then
+        assertEquals("Payment with id " + test_uuid + " not found", exception.getMessage());
+    }
+
+    @Test
+    void editMessageSucceeds() {
+        //when
+        paymentRepository.save(TEST_PAYMENT);
```
Comment (Evgenikaall) [inline src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java:98]: // given

File: src/test/java/com/endava/internship/mocking/service/BasicValidationServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        validationService = new BasicValidationService();
+    }
+
+    @Test
+    void validateValidAmount() {
+        //given
+        final Double validAmount = 124d;
+
+        //then
+        assertDoesNotThrow(() -> validationService.validateAmount(validAmount));
```
Comment (Evgenikaall) [inline src/test/java/com/endava/internship/mocking/service/BasicValidationServiceTest.java]: add message to assert

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java [hunk 5]
```diff
@@ -1,31 +1,104 @@
...
+    private PaymentRepository paymentRepository;
+
+    @Mock
+    private ValidationService validationService;
+
+    @InjectMocks
+    private PaymentService paymentService;
+
+    @Captor
+    ArgumentCaptor<Payment> argumentCaptor;
```
Comment (Evgenikaall) [inline src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: access modifier

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java [hunk 6]
```diff
@@ -1,31 +1,104 @@
...
+                "Amount should e equal to amount of payment was made before");
+        assertTrue(argumentCaptor.getValue().getMessage().contains(PETEA_USER.getName()));
 
     }
 
     @Test
-    void editMessage() {
+    void editMessageSucceeds() {
+        //when
+        when(paymentRepository.editMessage(PAYMENT.getPaymentId(), PAYMENT.getMessage())).thenReturn(PAYMENT);
```
Comment (Evgenikaall) [inline src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java:72]: // given

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java [hunk 7]
```diff
@@ -1,31 +1,104 @@
...
+
+    @Captor
+    ArgumentCaptor<Payment> argumentCaptor;
 
     @Test
-    void createPayment() {
+    void ShouldCreatePayment() {
+
+        //when
+        when(userRepository.findById(PETEA_USER.getId())).thenReturn(Optional.of(PETEA_USER));
```
Comment (Evgenikaall) [inline src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: // given

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java [hunk 8]
```diff
@@ -1,31 +1,104 @@
...
+                new Payment(2, 120d, "PAYMENT_2"),
+                new Payment(3, 104d, "PAYMENT_3"),
+                new Payment(4, 124804d, "PAYMENT_4")
+        );
+        final List<Payment> expectedPayments = new ArrayList<>();
+        expectedPayments.add(allPayments.get(0));
+        expectedPayments.add(allPayments.get(3));
+
+        //when
+        when(paymentRepository.findAll()).thenReturn(allPayments);
```
Comment (Evgenikaall) [inline src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: when must be under // given
is presteps for test

**Review by Evgenikaall — APPROVED** [review]
Have minor issues, but generally all good

File: src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java (hunk 1 above)
Comment (Valiusikk) [reply src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java]: Now exactly fixed!

File: src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java (hunk 2 above)
Comment (Valiusikk) [reply src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java:31]: Fixed!

File: src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java (hunk 3 above)
Comment (Valiusikk) [reply src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java:98]: Fixed!

File: src/test/java/com/endava/internship/mocking/service/BasicValidationServiceTest.java (hunk 4 above)
Comment (Valiusikk) [reply src/test/java/com/endava/internship/mocking/service/BasicValidationServiceTest.java]: Fixed!

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java (hunk 5 above)
Comment (Valiusikk) [reply src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: Fixed

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java (hunk 6 above)
Comment (Valiusikk) [reply src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java:72]: Fixed!

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java (hunk 7 above)
Comment (Valiusikk) [reply src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: fixed

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java (hunk 8 above)
Comment (Valiusikk) [reply src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: Fixed!

**Review by Evgenikaall — APPROVED** [review]
Looks good

**Pull request merged**